### DISCUSSION TITLE
"Use console's sound setting instead of being forced to Surround" Mod

### DISCUSSION THREAD

**Discussion by Bsquo** [leoetlino/project-restoration#193] [Q&A] [answered]
Hi.

Recently, I've created a small mod that removes a particular quirk regarding the audio in MM3D.
Basically, the game forces the sound output mode to always be "Surround" when in-game. This results in all of the audio sounding more compressed and in less quality overall (similarly to when hearing audio in low sample rates). All details (as well as the codes/patches) are available in the video below:

https://www.youtube.com/watch?v=ehZ1Eee3lPQ

The mod changes that and allows the game to output the audio in whatever sound setting it's set on the console. Notably, playing the mod with Stereo noticeably improves the audio quality.

I was thinking on making a pull request with this mod, but I'm a bit lost... When I was making the respective "hks" file (for the /hooks directory) for this patch, I thought... what patch address to which version should I add in the file? The address of the instruction I want to patch is different between the v100, v101 and v11 revisions, so I don't know which one should I put in the .hks file (I looked other .hks files, and those only have one address for each patch, although I don't know which version of the game is referring to specifically in those)
I'm pretty new to Magikoopa, so my apologies for the lengthy paragraph...

But even then, I also wanted to know if you considered this should belong to this project before proceeding myself. I personally think playing the game with Stereo is a very noticeable improvement in audio quality over Surround, but even better is letting the user select whatever sound output setting they want instead of being forced to just one...

Cheers!

**Comment by leoetlino** [comment, marked as answer]
Sure, making the game respect console settings sounds like a good idea. All addresses are for 1.0.0 (v100).

**Reply by Bsquo** [reply]
Thanks! I've just made the pull request
